packages/zwave-js/src/lib/node/Node.ts

```diff
@@ -5994,6 +5994,131 @@ ${formatRouteHealthCheckSummary(this.id, otherNode.id, summary)}`,
 		return true;
 	}
 
+	/**
+	 * Returns the current date, time and timezone (or a subset of those) on the node using one or more of the respective CCs.
+	 */
+	public async getDateAndTime(): Promise<object> {
```

Comment:
I suggest the following type as a return type (put it in `_Types.ts`) in the same directory:
```ts
export type DateAndTime = AllOrNone<{
	hour: number;
	minute: number;
	second: number;
}> &
	AllOrNone<{
		year: number;
		month: number;
		day: number;
	}> &
	AllOrNone<{
		dstOffset: number;
		standardOffset: number;
	}>;
```
This makes it clear that you'll get groups of properties, all time properties and/or all date properties and/or all timezone properties.

Reply:
> ```ts
> export type DateAndTime = AllOrNone<{
> 	hour: number;
> 	minute: number;
> 	second: number;
> }> &
> ```

Clock returns weekday instead of second.

Reply:
Then probably this:
```ts
export type DateAndTime = AllOrNone<{
  hour: number;
  minute: number;
}> &
  (
    | { weekday?: Weekday; second?: undefined }
    | { weekday?: undefined; second?: number }
  ) &
  AllOrNone<{
    year: number;
    month: number;
    day: number;
  }> &
  AllOrNone<{
    dstOffset: number;
    standardOffset: number;
  }>;
```